packages/jellyfish-api-core/__tests__/category/loan/auctionBid.test.ts

```diff
@@ -0,0 +1,350 @@
+      const node1AfterTSLAAmt = node1AfterTSLAAcc !== undefined ? Number(node1AfterTSLAAcc.split('@')[0]) : 0
+      expect(node1BeforeTSLAAmt - node1AfterTSLAAmt).toStrictEqual(525)
+
+      // end the auction and node1 get the auction bid
+      await tGroup.get(0).generate(36)
+      const auctionsAfter = await tGroup.get(1).container.call('listauctions')
+      expect(auctionsBefore[0].batches.length - auctionsAfter[0].batches.length).toStrictEqual(1)
+
+      const node1ColAccAfterWin = await tGroup.get(1).rpc.account.getAccount(node1ColAddr)
+      const wonBids = auctionsBefore[0].batches[0].collaterals
+      expect(node1ColAccAfterWin).toStrictEqual(wonBids.concat(node1ColAccAfter))
+    })
+  })
+
+  describe('test next auctionBid and utxos', () => {
+    beforeAll(async () => {
+      await tGroup.start()
+      await tGroup.get(0).container.waitForWalletCoinbaseMaturity()
+      await setup()
+
+      await tGroup.get(1).rpc.loan.auctionBid({
+        vaultId: vaultId,
+        index: 0,
+        from: node1ColAddr,
+        amount: '525@TSLA'
+      })
+      await tGroup.get(1).container.generate(1)
+      await tGroup.waitForSync()
+    })
+
+    afterAll(async () => {
+      await tGroup.stop()
+    })
+
+    it('next bid is required 1% higher', async () => {
+      const promise = tGroup.get(1).rpc.loan.auctionBid({
+        vaultId: vaultId,
+        index: 0,
+        from: node1ColAddr,
+        amount: '530@TSLA' // (525 * 1%) + 525 = 530.25
+      })
+
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow('Bid override should be at least 1% higher than current one')
+    })
+
+    it('should auctionBid with utxos', async () => {
+      const node1ColAccBefore = await tGroup.get(1).rpc.account.getAccount(node1ColAddr)
+      const node1BeforeTSLAAcc = node1ColAccBefore.length > 0
+        ? node1ColAccBefore.find((amt: string) => amt.split('@')[1] === 'TSLA')
+        : undefined
+      const node1BeforeTSLAAmt = node1BeforeTSLAAcc !== undefined ? Number(node1BeforeTSLAAcc.split('@')[0]) : 0
+
+      const utxo = await tGroup.get(0).container.fundAddress(node1ColAddr, 50)
+
+      const txid = await tGroup.get(1).rpc.loan.auctionBid({
+        vaultId: vaultId,
+        index: 0,
+        from: node1ColAddr,
+        amount: '550@TSLA' // (525 * 1%) + 525 = 530.25
+      }, [utxo])
+      expect(typeof txid).toStrictEqual('string')
+      await tGroup.get(1).container.generate(1)
+      await tGroup.waitForSync()
+
+      const node1ColAccAfter = await tGroup.get(1).rpc.account.getAccount(node1ColAddr)
+      const node1AfterTSLAAcc = node1ColAccAfter.length > 0
+        ? node1ColAccAfter.find((amt: string) => amt.split('@')[1] === 'TSLA')
+        : undefined
+      const node1AfterTSLAAmt = node1AfterTSLAAcc !== undefined ? Number(node1AfterTSLAAcc.split('@')[0]) : 0
+      expect(node1BeforeTSLAAmt - node1AfterTSLAAmt).toStrictEqual(25) // ?? not sure how it calculated yet
```

Comment:
I am also unsure. But I believe it is the amount value that you set in Line 319 minus the amount value that you set in Line 284. Is this a bug? I think it makes more sense if the value is `550`. Also, I think the C++ side shouldn't allow the same addresses to make another bid

Reply:
not bug.. it's correct... because its same bidder.. i updated the desc